File: Android_Busan_it/FirstProject/app/src/main/java/com/example/myapplication/MainActivity.java
package com.example.myapplication;

import android.os.Bundle;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

    EditText editText1;
    EditText editText2;

    Button button1;
    Button button2;
    Button button3;
    Button button4;
    Button button5;

    TextView textView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.practice_02_cal);

        editText1 = findViewById(R.id.EditView);
        editText2 = findViewById(R.id.EditView2);

        button1 = findViewById(R.id.button);
        button2 = findViewById(R.id.button2);
        button3 = findViewById(R.id.button3);
        button4 = findViewById(R.id.button4);
        button5 = findViewById(R.id.button5);

        textView = findViewById(R.id.textView3);

        button1.setOnClickListener(v -> {
            ButtonCall(button1, '+');
        });
        button2.setOnClickListener(v -> {
            ButtonCall(button2, '-');
        });
        button3.setOnClickListener(v -> {
            ButtonCall(button3, '*');
        });
        button4.setOnClickListener(v -> {
            ButtonCall(button4, '/');
        });
        button5.setOnClickListener(v -> {
            ButtonCall(button5, '%');
        });
    }

    private void makeToast(String str) {
        Toast.makeText(this, str, Toast.LENGTH_LONG).show();
    }

    private String calculate(String a, String b, char c) {
        long nTemp = 0;
        try {
            switch (c) {
                case '+': {
                    nTemp = Long.parseLong(a) + Long.parseLong(b);
                    break;
                }
                case '-': {
                    nTemp = Long.parseLong(a) - Long.parseLong(b);
                    break;
                }
                case '*': {
                    nTemp = Long.parseLong(a) * Long.parseLong(b);
                    break;
                }
                case '/': {
                    nTemp = Long.parseLong(a) / Long.parseLong(b);
                    break;
                }
                case '%': {
                    nTemp = Long.parseLong(a) % Long.parseLong(b);
                    break;
                }
                default: {
                    break;
                }
            }
        }catch (Exception e) {
            e.printStackTrace();
        }
        return String.valueOf(nTemp);
    }

    private void ButtonCall(Button btn, char c) {
        btn.setOnClickListener(v -> {
            if(editText1.getText().toString().equals("") || editText2.getText().toString().equals("")) {
                makeToast("EditBox 안에 값 입력 안했다!");
                return;
            }
            textView.setText("계산 결과: " + calculate(editText1.getText().toString(), editText2.getText().toString(), c));
        });
    }
}

File: Android_Busan_it/KotlinReMinder/app/src/main/java/com/example/kotlinreminder/MainActivity.kt
package com.example.kotlinreminder

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import android.widget.Toast

class MainActivity : AppCompatActivity() {
//    lateinit var edit1 : EditText
//    lateinit var edit2 : EditText
//    lateinit var btnAdd : Button
//    lateinit var btnSub : Button
//    lateinit var btnMul : Button
//    lateinit var btnDiv : Button
//    lateinit var textResult : TextView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        lateinit var edit1 : EditText
        lateinit var edit2 : EditText
        lateinit var btnAdd : Button
        lateinit var btnSub : Button
        lateinit var btnMul : Button
        lateinit var btnDiv : Button
        lateinit var btnPer : Button
        lateinit var textResult : TextView

        edit1 = findViewById(R.id.EditView)
        edit2 = findViewById(R.id.EditView2)
        btnAdd = findViewById<Button>(R.id.button)
        btnSub = findViewById<Button>(R.id.button2)
        btnMul = findViewById(R.id.button3)
        btnDiv = findViewById(R.id.button4)
        btnPer = findViewById(R.id.button5)
        textResult = findViewById(R.id.textView3)

        callListener(btnAdd, edit1, edit2, '+', textResult)
        callListener(btnSub, edit1, edit2, '-', textResult)
        callListener(btnMul, edit1, edit2, '*', textResult)
        callListener(btnDiv, edit1, edit2, '/', textResult)
        callListener(btnPer, edit1, edit2, '%', textResult)
    }

    private fun callListener(btn : Button, edit1: EditText, edit2: EditText, c: Char, result: TextView) {
        btn.setOnClickListener { v ->
            var str1 = edit1.text.toString()
            var str2 = edit2.text.toString()

            if(str1 == "" || str2 == "") {
                makeToast("빈 칸을 채워주세요!")
                return@setOnClickListener
            }

            var a = str1.toInt()
            var b = str2.toInt()
            var cal = when(c) {
                '+' -> a + b
                '-' -> a - b
                '*' -> a * b
                '/' -> a / b
                '%' -> a % b
                else -> 0
            }
            result.text = cal.toString()
        }
    }

    private fun makeToast(str : String) {
        Toast.makeText(this, str, Toast.LENGTH_LONG).show()
    }
}